src/uproot/models/RNTuple.py

```diff
@@ -323,22 +378,48 @@ def arrays(
         L = clusters[0].num_entries
 
         form = self.to_akform().select_columns(filter_names)
+        # only read columns mentioned in the awkward form
+        target_cols = []
         D = {}
+        _recursive_find(form, target_cols)
         for i, cr in enumerate(self.column_records):
-            n = cr.field_id
-            D[f"field-{n}"] = self.read_col_page(i, L)
+            key = f"column-{i}"
+            if key not in target_cols:
+                continue
+
+            content = self.read_col_page(i, L)
+            if cr.type == 3:
+                # split Switch column into index and tag
+                # TODO what's this -1 hack
+                tags = (content >> 44).astype("int8") - 1
+                kindex = numpy.bitwise_and(content, numpy.int64(0x00000000000FFFFF))
+                D[f"{key}-index"] = kindex
+                D[f"{key}-tags"] = tags
+            else:
+                # don't distinguish data and offsets
+                D[f"{key}-data"] = content
+                D[f"{key}-offsets"] = content
```

Comment:
With a small set of exceptions, I'm generally against using `break` and `continue` to express logic (as well as early `return` from functions, unless it's handling a special case). These keywords are the last vestiges of `goto`.

(Note: [I have used goto](https://github.com/scikit-hep/awkward/blob/af0f4569ed90eab64690098e5a64ae610282c4bf/src/libawkward/forth/ForthMachine.cpp#L3680-L3685); there are exceptions for most of these rules. Generally, these Bad Things should just be used sparingly.)

```suggestion
            if key in target_cols:
                content = self.read_col_page(i, L)
                if cr.type == 3:
                    # split Switch column into index and tag
                    # TODO what's this -1 hack
                    tags = (content >> 44).astype("int8") - 1
                    kindex = numpy.bitwise_and(content, numpy.int64(0x00000000000FFFFF))
                    D[f"{key}-index"] = kindex
                    D[f"{key}-tags"] = tags
                else:
                    # don't distinguish data and offsets
                    D[f"{key}-data"] = content
                    D[f"{key}-offsets"] = content
```

Also, that `44` and `0x00000000000FFFFF` either need to be named constants _or_ the two lines that split tags and index should be a short, named function. It's all about isolating the "`44`" and "`0x00000000000FFFFF`" and explaining what they do, so they can still be magic numbers if they're inside a well-named function that does only this one thing.

Reply:
In this case I did the original way to avoid indenting many lines without an else branch

Reply:
Right, that's understandable. The indentation as syntax thing fails for long blocks, and these are a bit long. (There are longer ones elsewhere in Uproot...) An editor that displays vertical lines at each indentation depth can help, though admittedly I don't use one.